src/component.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Base class for all MDL components.
+ *
+ * @export
+ */
+class MaterialComponent { // eslint-disable-line no-unused-vars
+  /**
+   * Initialize component from a DOM node.
+   *
+   * @param {Element} root The element being upgraded.
+   */
+  constructor(root) {
+    this.root_ = root;
+
+    // Check if the root has the right class.
+    if (!root.classList.contains(this.constructor.classes_.ROOT)) {
+      throw new Error(`${this.constructor.classAsString} missing ` +
+          `${this.constructor.classes_.ROOT} class.`);
+    }
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * CSS classes used in this component.
+   *
+   * @protected
+   * @return {Object<string, string>} The CSS classes used in this component.
+   */
+  static get classes_() {
```

Comment:
Just wondering: `CssClasses_()` rather? `classes_()` seems dangerous considering the ambiguity.